src/Microsoft.Fast.Components.FluentUI/Components/InputFile/FluentInputFileEventArgs.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace Microsoft.Fast.Components.FluentUI;
+
+/// <summary />
+public class FluentInputFileEventArgs : EventArgs
+{
+    /// <summary>
+    /// Gets the index of the current file in an upload process.
+    /// </summary>
+    public int Index { get; internal set; } = 0;
+
+    /// <summary>
+    /// Gets the local file of the current file in an upload process.
+    /// Only if Mode = SaveToTemporaryFolder (otherwise, this value is null).
+    /// </summary>
+    public FileInfo? LocalFile { get; internal set; }
+
+    /// <summary>
+    /// Gets a small buffer data of the current file in an upload process.
+    /// Only if Mode = Buffer.
+    /// </summary>
+    public FluentInputFileBuffer Buffer { get; internal set; } = default!;
+
+    /// <summary>
+    /// Gets a reference to the current stream in an upload process.
+    /// Only if Mode = Stream (otherwise, this value is null).
+    /// The OnProgressChange event will not be triggered.
+    /// </summary>
+    public Stream? Stream { get; internal set; }
+
+    /// <summary>
+    /// Gets the name of the current file in an upload process.
+    /// </summary>
+    public string Name { get; internal set; } = string.Empty;
+
+    /// <summary>
+    /// Gets the size (in bytes) of the current file in an upload process.
+    /// </summary>
+    public long Size { get; internal set; } = 0;
+
+    /// <summary>
+    /// Gets the content type of the current file in an upload process.
+    /// </summary>
+    public string ContentType { get; internal set; } = string.Empty;
+
+    /// <summary>
+    /// Gets the label to display in an upload process.
+    /// </summary>
+    public string ProgressTitle { get; internal set; } = string.Empty;
+
+    /// <summary>
+    /// Gets the global percent value in an upload process.
+    /// </summary>
+    public int ProgressPercent { get; internal set; } = 0;
+
+    /// <summary>
+    /// Gets the error message (or null if no error occurred).
+    /// </summary>
+    public string? ErrorMessage { get; internal set; }
+
+    /// <summary>
+    /// Gets a list of all files currently in an upload process.
+    /// </summary>
+    public IEnumerable<(string Name, long Size, string ContentType)> AllFiles { get; internal set; } = default!;
```

Comment:
Could you avoid to define public anonymous classes? And create a "real" explicit class for this part?